packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/samples-dev/dataProductsAddUserRoleSample.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { NetworkAnalyticsClient } from "@azure/arm-networkanalytics";
+import { DefaultAzureCredential } from "@azure/identity";
+
+/**
+ * This sample demonstrates how to assign role to the data product.
+ *
+ * @summary assign role to the data product.
+ * x-ms-original-file: 2023-11-15/DataProducts_AddUserRole_MaximumSet_Gen.json
+ */
+async function dataProductsAddUserRoleMaximumSetGen() {
```

Comment:
One limitation of tshy is even the "includes" in tsconfig.json contains test/samples-dev but we still can't inlcude their compile. This would be hard for us to detect any generation issue in samples/test.

We plan to have a [command ](https://github.com/Azure/azure-sdk-for-js/pull/30810)in dev-tool to help check in SDK repo. maybe not in recent priority but I feel like we should also include similar check in codegen repo.

@joheredi @wanlwanl @qiaozha how do you think?